tests/spark/rflx-arrays_tests.adb

```diff
@@ -27,898 +27,414 @@ package body RFLX.Arrays_Tests is
-               pragma Loop_Invariant (Sequence_Context.Last = Sequence_Context.Last'Loop_Entry);
-               pragma Loop_Invariant (not Arrays.Inner_Message.Has_Buffer (Element_Context));
-
-               Arrays.Inner_Messages.Switch (Sequence_Context, Element_Context);
-
-               Arrays.Inner_Message.Verify_Message (Element_Context);
-               if Arrays.Inner_Message.Valid (Element_Context, Arrays.Inner_Message.F_Length) then
-                  Length := Arrays.Inner_Message.Get_Length (Element_Context);
-                  Assert (Length'Image, I'Image, "Unexpected Length of element " & I'Image);
-               end if;
-               Assert (Arrays.Inner_Message.Structural_Valid_Message (Element_Context),
-                       "Structural invalid element " & I'Image);
-
-               Arrays.Inner_Messages.Update (Sequence_Context, Element_Context);
-
-               I := I + 1;
-            end loop;
-
-            Assert (I'Image, Natural'Image (3), "Unexpected number of elements");
-
-            Assert (not Arrays.Messages_Message.Valid (Context, Arrays.Messages_Message.F_Messages),
-                    "Valid Messages before context update");
-            Arrays.Messages_Message.Update_Messages (Context, Sequence_Context);
-            Assert (Arrays.Messages_Message.Valid (Context, Arrays.Messages_Message.F_Messages),
-                    "Invalid Messages after context update");
-         else
-            Assert (False, "Invalid Messages or Buffer");
-         end if;
-      else
-         Assert (False, "Invalid Length");
-      end if;
-
-      Assert (Arrays.Messages_Message.Valid_Message (Context), "Invalid Message after complete parsing");
-
-      if Arrays.Messages_Message.Has_Buffer (Context) then
-         Arrays.Messages_Message.Take_Buffer (Context, Buffer);
-      end if;
-      Free_Bytes_Ptr (Buffer);
+         Message.Switch_To_Modular_Vector (Context, Sequence_Context);
+
+         while I <= 10 and then Arrays.Modular_Vector.Has_Element (Sequence_Context) loop
+            pragma Loop_Invariant (Arrays.Modular_Vector.Has_Buffer (Sequence_Context));
+            pragma Loop_Invariant (Context.Buffer_First = Sequence_Context.Buffer_First);
+            pragma Loop_Invariant (Context.Buffer_Last = Sequence_Context.Buffer_Last);
+            pragma Loop_Invariant (Sequence_Context.First = Sequence_Context.First'Loop_Entry);
+            pragma Loop_Invariant (Sequence_Context.Last = Sequence_Context.Last'Loop_Entry);
+
+            Arrays.Modular_Vector.Next (Sequence_Context);
+
+            Assert (Arrays.Modular_Vector.Valid_Element (Sequence_Context), "Invalid element " & I'Image);
+
+            Element := Arrays.Modular_Vector.Get_Element (Sequence_Context);
+
+            Assert (Element'Image, Natural'Image (I), "Invalid value of element " & I'Image);
+
+            I := I + 1;
+         end loop;
```

Comment:
Is the array meant to have only 10 elements? If so it might make sense to move the `I <= 10` check to an assertion to detect too many elements.

Reply:
Changed in 354c5f3.